### DISCUSSION TITLE
ESTOP: uCNC "pin" vs. pin on Arduino board vs. pin on CNC Shield V.3 - a documentation question

### DISCUSSION THREAD

**Discussion by hmmueller** [Paciente8159/uCNC#708] [Q&A] [answered]
(re-)Starting my µCNC+UGS journey, I immediately stumble over a documentation problem with the "ESTOP pin" - please bear with me, I'm very happy about µCNC, but it's a little bit hard going for me at the moment:

The uCNC documents talk about "pins" - e.g. [this important text](https://github.com/Paciente8159/uCNC/wiki/Basic-user-guide#%C2%B5cnc-wiring) starts with

> By default, and as a safety measure, µCNC control pins (Emergency stop, Safety door and feed hold) and all limit switch pins

Now, there is no Emergency stop or Safety door pin on the Arduino; so there must be a mapping of those "uCNC names" to Arduino pins. 

Ok, there is [the HAL mapping page](https://github.com/Paciente8159/uCNC/wiki/ALL-MCU:-Customizing) (I had to learn that HAL is hardware abstraction layer - newbie, sorry!), which starts with

> Taking the boardmap_grbl.h file inside src\hal\boards\avr folder ...

Unfortunately, there is no boardmap_grbl.h file there - neither in the master nor the 1.7.2 I'm using. So, using a CNC V.3 Shield, I open the file boardmap_uno_shield_v3.h ... which #includes boardmap_uno.h.

There, I finally find a definition for the virtual ESTOP pin:

> #define ESTOP_BIT 0

and, somewhere below (which I had first overlooked)

> #define ESTOP_PORT C

In the UNO [diagram for AVR Customizing](https://github.com/Paciente8159/uCNC/wiki/AVR:-Customizing), I now find a pin "Port pin" called PC0 (and now I understand those PORT values of D, C or B ...). So, µCNC's ESTOP is A0 on the Arduino.

Now, where is this on the CNC Shield V.3? Unfortunately, in [Protoneer's schematic for the V.3 CNC Shield](https://blog.protoneer.co.nz/wp-content/uploads/2013/07/Arduino-CNC-Shield-Scematics-V3.XX_.jpg), the connector name is totally unreadable for me. Hm. But there is another diagram around, e.g. [here](https://forum.arduino.cc/t/using-cnc-shield-v3-directly-with-arduino-ide/1022999/3) - and this one says that A0 goes to the "Abort" pin on the shield.

**So my question is**: Is it correct to interpret µCNC's "Short the ESTOP pin to ground" as "Connect Abort on the CNC Shield V.3 to ground"?
And if not: Where did I run astray in my documentation journey??

By the way, it is unfortunate that there is also an E-STOP pin (note the hyphen) on the CNC Shield V.3, which has nothing to do with µCNC's virtual ESTOP pin - it is connected to Arduino's RST pin; and must "of course" _not_ be grounded - which I, again of course, did at first - and nothing worked any more :-P

Thanks a lot for any help and enlightenment!
Harald M.

**Comment by Paciente8159** [comment, marked as answer]
About UNO pins. One thing is the Arduino pin names. The other is the chip pin name.
Take a look at the [UNO pinout](https://github.com/Paciente8159/uCNC/blob/master/docs/arduino-uno-pinout-diagram.svg)

The Arduino pin A0 is the chip pin C0.
This is because uCNC was originally designed to be compiled independently of Arduino framework.

If you don't have control pins hardwired you can disable them all along, or disable them individually simply by undefining them.

**Reply by hmmueller** [reply]
Thanks a lot for quick answer. 

It's just that I have to translate twice: µCNC --> Arduino, and then Arduino --> CNC Sketch name ... but I think I know now how it works.

I actually do (or should have) control of all the hardware pins - I can use jumpers to nicely ground them for the moment. Most probably, I'll come back to them later when the motors start moving :-)

H.M.
